packages/x-charts/src/ChartsSurface/ChartsSurface.tsx

```diff
@@ -44,9 +47,14 @@ const ChartChartsSurfaceStyles = styled('svg', {
 
 const ChartsSurface = React.forwardRef<SVGSVGElement, ChartsSurfaceProps>(function ChartsSurface(
   inProps: ChartsSurfaceProps,
+  // What to do with ref coming from forwardRef and Context?
   ref: React.Ref<SVGSVGElement>,
 ) {
-  const props = useThemeProps({ props: inProps, name: 'MuiChartsSurface' });
+  const contextProps = useSurfaceProps();
+  const resolvedProps = resolveProps(inProps as any, contextProps) as ChartsSurfaceProps;
```

Comment:
Do we need to have the width/height in the props?

The one in the context should always be defined and will then override the `inProps`.

What about removing width, height, and viewBox from the props and only consider them as comming from the DataProvider to prevent users from trying to modify them

Reply:
I've removed the viewport, seems unnecessary since we are forcing the width/height already. Do you think it is necessary somehow?